src/inference/src/dev/itensor.cpp

```diff
@@ -0,0 +1,490 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "dev/itensor.hpp"
+
+#include "ie_blob.h"
+#include "ie_ngraph_utils.hpp"
+#include "ie_remote_blob.hpp"
+#include "openvino/core/except.hpp"
+#include "openvino/runtime/allocator.hpp"
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+
+const Shape& ITensor::get_shape() const {
+    OPENVINO_NOT_IMPLEMENTED;
+}
+
+const element::Type& ITensor::get_element_type() const {
+    OPENVINO_NOT_IMPLEMENTED;
+}
```

Comment:
Could you please elaborate on why these methods are implemented with `OPENVINO_NOT_IMPLEMENTED`, is not it better to make them pure methods? This technik sometimes used when such calls are not mandatory or senseless in the derivative types, but it looks like these two should be implemented in all the successors.